spacy/kb/kb.pxd

```diff
@@ -0,0 +1,10 @@
+"""Knowledge-base for entity or concept linking."""
+
+from cymem.cymem cimport Pool
+from libc.stdint cimport int64_t
+from ..vocab cimport Vocab
```

Comment:
Isn't using a `Vocab` an implementation detail of the current implementation that may well not be shared by other implementations?

Reply:
Hm, that's true. There isn't really a need for it in the base class. @svlandeg Any objections to moving `vocab` to `InMemoryLookupKB`?

Reply:
We should do it at some point (latest by v4) but I'm not sure we want to do it now, because it'll introduce additional complexity in `Candidate` not being able to rely on `kb.vocab` and then we'll have to end up changing more public APIs ... so maybe better to do this later?

Reply:
Right, I forgot `Candidate` uses `vocab`. No, let's leave it in. Added it to the list of v4 changes.